### PR TITLE
API translations in core

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

**Comment by pandamicro** [comment]
https://github.com/cocos-creator/3d-tasks/issues/2506

**Comment by pandamicro** [comment]
@shrinktofit please review some ts type changes

File: cocos/particle/particle-utils.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class ParticleUtils {
         }
         if (!this.particleSystemPool.has(prefab._uuid)) {
             this.particleSystemPool.set(prefab._uuid, new Pool<CCObject>(() => {
-                return instantiate(prefab);
+                return instantiate(prefab) || new Node();
```
Comment (pandamicro) [inline cocos/particle/particle-utils.ts:24]: @caryliu1999 is this acceptable ? instantiate could fail

File: cocos/core/data/instantiate.ts [hunk 2]
```diff
@@ -63,7 +62,7 @@ const objsToClearTmpVar: any = [];   // used to reset _iN$t variable
  * node.parent = scene;
  * ```
  */
-function instantiate (original, internal_force?) {
+function instantiate (original:Prefab|Node|Object, internal_force?:Node|Object): Node | null {
```
Comment (shrinktofit) [inline cocos/core/data/instantiate.ts]: 空格

File: cocos/core/data/object.ts [hunk 3]
```diff
@@ -250,7 +245,7 @@ class CCObject {
      * cc.log(node.isValid);    // false, destroyed in the end of last frame
      * ```
      */
-    get isValid () {
+    get isValid ():boolean {
```
Comment (shrinktofit) [inline cocos/core/data/object.ts]: 这种也得空格呀

File: cocos/core/data/object.ts [hunk 4]
```diff
@@ -400,7 +395,7 @@ if (EDITOR) {
  */
 // @ts-ignore
 prototype._deserialize = null;
-/**
+/*
```
Comment (shrinktofit) [inline cocos/core/data/object.ts:398]: 这是为啥？

**Comment by pandamicro** [comment]
Rebased to https://github.com/cocos-creator/engine/pull/6571/ and made some imporvement

**Review by shrinktofit — APPROVED** [review]

File: cocos/core/data/object.ts (hunk 4 above)
Comment (pandamicro) [reply cocos/core/data/object.ts:398]: 私有方法不用暴露 API 文档

**Comment by pandamicro** [comment]
@shrinktofit 格式优化已更新

File: cocos/particle/particle-utils.ts (hunk 1 above)
Comment (caryliu1999) [reply cocos/particle/particle-utils.ts:24]: > is this acceptable ? instantiate could fail

Yes, this method is not used currently.

**Review by caryliu1999 — APPROVED** [review]

**Pull request merged**